### PR TITLE
[KOGITO-2889] Build multiple jars for different persistence of data index

### PR DESCRIPTION
**Pull request by xieshenzh**
Break the `data-index-service` module to three new modules: `data-index-service-common`, `data-index-service-infinispan`, `data-index-service-mongodb`. 
Two uberJars will be created for `data-index-service-infinispan` and `data-index-service-mongodb`.

Jira: https://issues.redhat.com/browse/KOGITO-2889

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by xieshenzh** [comment]
Assembly:
https://github.com/kiegroup/kogito-apps/pull/350
https://github.com/kiegroup/kogito-runtimes/pull/658

File: data-index/data-index-service/data-index-service-mongodb/pom.xml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <configuration>
+          <excludes>
+            <exclude>**/*Native*IT.java</exclude>
+          </excludes>
+          <systemProperties>
+            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+            <container.image.infinispan>${container.image.infinispan}</container.image.infinispan>
```
Comment (danielezonca) [inline data-index/data-index-service/data-index-service-mongodb/pom.xml]: Can you please verify if it is needed?

**Review by danielezonca — APPROVED** [review]
LGTM, just a minor comment

File: data-index/data-index-service/data-index-service-mongodb/pom.xml (hunk 1 above)
Comment (xieshenzh) [reply data-index/data-index-service/data-index-service-mongodb/pom.xml]: It is not needed. I have removed this line. Thanks a lot.

**Comment by cristianonicolai** [comment]
@xieshenzh can you please review the conflicts, thanks

**Comment by xieshenzh** [comment]
Jenkins retest this

**Comment by xieshenzh** [comment]
Jenkins retest this

**Comment by xieshenzh** [comment]
Jenkins retest this

**Comment by xieshenzh** [comment]
> @xieshenzh can you please review the conflicts, thanks

@cristianonicolai conflicts resolved. Thanks.

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by cristianonicolai — APPROVED** [review]

**Comment by xieshenzh** [comment]
@ricardozanini could you please take a look at this PR? 

Mainly, the change is to break the `data-index-service` module to three modules: 
- `data-index-service-common` contains code which is independent of storage type
- `data-index-service-infinispan` is for building the jar that works with infinispan 
- `data-index-service-mongodb` is for mongodb. 

Mostly, code is just moved to another module, not changed. 

Thanks.

**Review by ricardozanini — APPROVED** [review]
LGTM

**Comment by ricardozanini** [comment]
@xieshenzh Sonar is complaining on the test coverage. Are we ok with 54%? @cristianonicolai @danielezonca let me know so I can merge this.

**Comment by danielezonca** [comment]
Fine for me to merge, this PR is mainly moving existing code :+1: from my side

**Pull request merged**